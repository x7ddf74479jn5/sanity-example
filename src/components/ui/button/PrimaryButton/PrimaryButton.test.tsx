import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import renderer from "react-test-renderer";

import { PrimaryButton } from ".";

describe("ui/buttons/PrimaryButton", () => {
  describe("Snapshot", () => {
    it("button", () => {
      const component = renderer.create(<PrimaryButton tag="button">label</PrimaryButton>);
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot("button");
    });

    it("link", () => {
      const component = renderer.create(
        <PrimaryButton tag="a" linkProps={{ href: "/" }}>
          label
        </PrimaryButton>
      );
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot("link");
    });

    it("input", () => {
      const component = renderer.create(<PrimaryButton tag="input" value="OK" type="button" />);
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot("input");
    });
  });

  it("button", () => {
    const handleClick = jest.fn();
    render(
      <PrimaryButton tag="button" onClick={handleClick}>
        label
      </PrimaryButton>
    );
    const button = screen.getByRole("button", { name: "label" });
    expect(button).toBeEnabled();
    userEvent.click(button);
    expect(handleClick).toBeCalled();
  });

  it("link", () => {
    render(
      <PrimaryButton tag="a" linkProps={{ href: "/" }}>
        label
      </PrimaryButton>
    );
    const button = screen.getByRole("button", { name: "label" });
    expect(button).toHaveAttribute("href", "/");
  });

  it("input", () => {
    render(<PrimaryButton tag="input" value="OK" type="button" />);
    const button = screen.getByRole("button");
    expect(button).toBeEnabled();
    expect(button).toHaveValue("OK");
  });
});
