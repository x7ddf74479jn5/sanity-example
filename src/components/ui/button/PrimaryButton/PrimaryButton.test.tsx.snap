// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui/buttons/PrimaryButton Snapshot button: button 1`] = `
<button
  type="button"
>
  label
</button>
`;

exports[`ui/buttons/PrimaryButton Snapshot input: input 1`] = `
<input
  type="button"
  value="OK"
/>
`;

exports[`ui/buttons/PrimaryButton Snapshot link: link 1`] = `
<a
  href="/"
  onClick={[Function]}
  onMouseEnter={[Function]}
  role="button"
>
  label
</a>
`;
